components/core/tools/docker-images/clp-env-base-ubuntu-focal/Dockerfile

```diff
@@ -1,18 +1,30 @@
-FROM ubuntu:focal
+FROM ubuntu:focal AS BASE
 
 WORKDIR /root
 
 RUN mkdir -p ./tools/scripts/lib_install
 ADD ./tools/scripts/lib_install ./tools/scripts/lib_install
 
-RUN ./tools/scripts/lib_install/ubuntu-focal/install-all.sh
+# NOTE: We can't use the `install-all.sh` script since we need to configure the compiler in between
+# `install-prebuilt-packages.sh` and `install-packages-from-source.sh`.
+RUN ./tools/scripts/lib_install/ubuntu-focal/install-prebuilt-packages.sh
 
-# Set the compiler to gcc-10
-RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-10 10
-RUN update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-10 10
-RUN update-alternatives --set gcc /usr/bin/gcc-10
-RUN update-alternatives --set g++ /usr/bin/g++-10
+# Set gcc/g++ 10 as the default compiler
```

Comment:
I actually feel the original way we wrote it is more readable. is there a reason why you turn it into a one line command?

Reply:
I was trying to reduce the number of generated layers since each `RUN` command generates a separate layer; but I guess we can keep them separate since we're flattening the image in the end.